packages/hardware-ledger/src/LedgerKeyAgent.ts

```diff
@@ -120,16 +120,33 @@ const isMultiSig = (tx: Transaction): boolean => {
   return result;
 };
 
+type LedgerConnectionWithCommunicationTypeAndDevicePath = {
+  deviceConnection: LedgerConnection;
+  communicationType: CommunicationType;
+  devicePath?: string;
+};
+
 export class LedgerKeyAgent extends KeyAgentBase {
   readonly deviceConnection?: LedgerConnection;
   readonly #communicationType: CommunicationType;
+  static deviceConnections: LedgerConnectionWithCommunicationTypeAndDevicePath[] = [];
 
   constructor({ deviceConnection, ...serializableData }: LedgerKeyAgentProps, dependencies: KeyAgentDependencies) {
     super({ ...serializableData, __typename: KeyAgentType.Ledger }, dependencies);
     this.deviceConnection = deviceConnection;
     this.#communicationType = serializableData.communicationType;
   }
 
+  static findKeyAgentByCommunicationTypeAndDevicePath(
+    communicationType: CommunicationType,
+    devicePath?: string,
+    deviceConnections?: LedgerConnectionWithCommunicationTypeAndDevicePath[]
```

Comment:
I do not understand the need to pass this as a param. Shouldn't it use the static one availble? `LedgerKeyAgent.deviceConnections`

Reply:
Good point, i'll update. Thanks for catching this

Reply:
Updated in [99891ec](https://github.com/input-output-hk/cardano-js-sdk/pull/885/commits/99891ec44752b799256cfd0a78d910b3f0589307)